### PR TITLE
perf(admin): replace full-ID count query with SELECT COUNT(*) for AI orders pagination

### LINKED ISSUE #230: perf(admin): replace full-ID count query with SELECT COUNT(*) for recent-orders pagination
## Problem

The `/admin/recent-orders` REST endpoint runs two queries per paginated request:

1. `wc_get_orders( $query_args )` — fetches the current page of order objects (e.g. 10 rows). ✅ Efficient.
2. `wc_get_orders( [ 'limit' => -1, 'return' => 'ids', ... ] )` — fetches **all matching order IDs** just to call `count()` on the result for pagination metadata. ❌ Scales with total matching orders, not page size.

On a store with 50k+ AI-attributed orders, the second query scans the full result set, allocates a large ID array in PHP, transfers it over the DB connection, then immediately discards it after `count()`. This happens on every page turn in the AI Orders table.

## Fix

Replace the count query with a direct `SELECT COUNT(*)` via `$wpdb->get_var()`, applying the same filters (status, meta_query, search, agent) hand-translated to SQL. The DB computes the total internally and returns a single integer — no data transfer, no PHP array.

## Acceptance criteria

- `get_recent_orders()` issues a `SELECT COUNT(*)` for the total instead of `wc_get_orders( [ 'limit' => -1 ] )`
- The count query applies the same status/meta/search/agent filters as the main query
- Unit test covers the count path
- No regression on the paginated response shape

## References

Flagged by Copilot in PR #229. Deferred from that PR as out of scope.

### PR DESCRIPTION
**Pull request by pierorocca**
## Summary

- Adds `count_ai_orders()` private method to `WC_AI_Storefront_Admin_Controller` that issues a single `SELECT COUNT(DISTINCT id)` query instead of fetching all matching order IDs into PHP memory
- Handles both HPOS (`wc_orders`/`wc_orders_meta`) and legacy (`wp_posts`/`wp_postmeta`) storage, applying the same status/agent/search filters as the main page query
- Sets up `$wpdb` mock in `AdminRecentOrdersTest::setUp()` so all 22 existing tests continue to pass; adds 2 new assertions covering the `total` field and the null-cast edge case

## Why

`wc_get_orders( limit=-1, return='ids' )` loads every matching order ID into PHP memory to call `count()` on the array. On stores with thousands of AI-attributed orders this is wasteful — a `SELECT COUNT(*)` returns one integer from the DB at negligible cost.

Closes #230

## Test plan

- [ ] `vendor/bin/phpunit` — 1008/1008 pass
- [ ] `vendor/bin/phpcs includes/admin/class-wc-ai-storefront-admin-controller.php` — clean
- [ ] `npx wp-scripts lint-js client/` — clean

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: includes/admin/class-wc-ai-storefront-admin-controller.php [hunk 1]
```diff
@@ -745,6 +740,159 @@ public function get_recent_orders( $request ) {
...
+			if ( $agent_filter ) {
+				$sql     .= ' AND m.meta_value = %s';
+				$params[] = $agent_filter;
+			}
+
+			if ( $search ) {
+				// billing_first_name and billing_last_name live in wc_order_addresses
+				// (not on wc_orders directly). Join once with address_type='billing'
+				// and reference a.first_name / a.last_name; billing_email IS a direct
+				// column on wc_orders so no join needed for that field.
```
[the remark this answers is not in the document]
Comment (pierorocca) [reply includes/admin/class-wc-ai-storefront-admin-controller.php]: Fixed in 2d29ecd — comment updated to describe the correlated EXISTS subqueries instead of a join.

File: tests/php/unit/AdminRecentOrdersTest.php [hunk 2]
```diff
@@ -334,4 +359,38 @@ public function test_wc_get_orders_wp_error_returns_empty_orders_payload(): void
...
+		$data     = $response->get_data();
+
+		$this->assertSame( 42, $data['total'] );
+		$this->assertCount( 1, $data['orders'] );
+	}
+
+	public function test_count_ai_orders_returns_zero_when_db_returns_null(): void {
+		// $wpdb->get_var() returns null on SQL error or when no rows match.
+		// count_ai_orders() must cast null → 0 (int) so the JSON response
+		// has a valid integer in the `total` field.
```
[the remark this answers is not in the document]
Comment (pierorocca) [reply tests/php/unit/AdminRecentOrdersTest.php]: Fixed in 2d29ecd — comment updated to clarify that `COUNT(*)` always returns a row (even if 0), so null from `get_var()` represents a DB error, not a no-match result.
